File: Determine_if_String_Halves_Are_Alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        left = left_vowels = right_vowels = 0
        right = len(s) - 1
        vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}

        while left < right:
            if s[left] in vowels:
                left_vowels += 1
            if s[right] in vowels:
                right_vowels += 1
            
            left += 1
            right -= 1
        
        return left_vowels == right_vowels

File: Determine_if_String_Halves_Are_Alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s) {
        int left = 0, leftVowels = 0, rightVowels = 0;
        int right = s.length() - 1;
        unordered_set<char> vowels {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};

        while (left < right) {
            if (vowels.count(s[left])) {
                leftVowels++;
            }
            if (vowels.count(s[right])) {
                rightVowels++;
            }

            left++;
            right--;
        }

        return leftVowels == rightVowels;        
    }
};
